### DISCUSSION TITLE
Misleading example of request-scoped data loaders?

### DISCUSSION THREAD

**Discussion by breun** [graphql-java-kickstart/graphql-spring-boot#191] [Q&A]
https://github.com/graphql-java-kickstart/graphql-java-servlet/blob/master/README.md#request-scoped-dataloaders shows an example of how to use request-scoped data loaders. However, looking at the `buildDataLoaderRegistry()` method it seems that a new `DataLoaderRegistry` is created for each request, but the exact same `userDataLoader` is put in, so as far as I understand the data loader is still shared between requests (including its cache):

```
private DataLoaderRegistry buildDataLoaderRegistry() {
    DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
    dataLoaderRegistry.register("userDataLoader", userDataLoader);
    return dataLoaderRegistry;
}
```

The example at https://github.com/graphql-java-kickstart/graphql-spring-boot/blob/master/example-request-scoped-dataloader/src/main/java/graphql/servlet/examples/dataloader/requestscope/CustomGraphQLContextBuilder.java does create a new `DataLoader` instance for every call to `buildDataLoaderRegistry()`, which seems right to me:

```
private DataLoaderRegistry buildDataLoaderRegistry() {
    DataLoaderRegistry dataLoaderRegistry = new DataLoaderRegistry();
    dataLoaderRegistry.register("customerDataLoader",
            new DataLoader<Integer, String>(customerIds ->
                    CompletableFuture.supplyAsync(() ->
                            customerRepository.getUserNamesForIds(customerIds))));
    return dataLoaderRegistry;
}
```

Am I correct in thinking the first example should be adjusted to also create new instances for every call to `buildDataLoaderRegistry()`?

**Comment by oliemansm** [comment]
@breun Ah, you're correct. Thanks for pointing it out, will update it.
